cmd/argo/main.go

```diff
@@ -5,6 +5,8 @@ import (
 	"os"
 
 	"github.com/argoproj/argo/cmd/argo/commands"
+	// load the gcp plugin (required to authenticate against GKE clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```

Comment:
?? Why does gcp need special plugins? Will similar such plugins be required for other K8s clusters?

Reply:
It looks like they separate out this way as to "not overloading regular clients with unnecessary imports. E.g you don't have to import OpenID Connect and Azure packages"

https://github.com/kubernetes/client-go/issues/242#issuecomment-314662652